target/linux/ramips/dts/mt7620a_engenius_esr600.dts

```diff
@@ -0,0 +1,211 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include "mt7620a.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "engenius,esr600", "ralink,mt7620a-soc";
+	model = "EnGenius ESR600";
+
+	memory@0 {
+		device_type = "memory";
+		reg = <0x0 0x4000000>;
+	};
+
+	chosen {
+		bootargs = "console=ttyS0,115200";
+	};
+
+	aliases {
+		led-boot = &led_power;
+		led-failsafe = &led_power;
+		led-running = &led_power;
+		led-upgrade = &led_power;
+	};
+
+	leds {
+		// FIXME: locate WPS 5g blue LED
+		// ISSUE: taking the WAN LED on gpio2/5 
+		// causes AR8327 switch to fail initialization
+
+		compatible = "gpio-leds";
+
+		led_power: power {
+			label = "esr600:amber:power";
+			gpios = <&gpio0 9 GPIO_ACTIVE_LOW>;
+		};
+		wps2g {
+			label = "esr600:amber:wps2g";
+			gpios = <&gpio2 6 GPIO_ACTIVE_LOW>;
+		};
+		wlan5g {
+			label = "esr600:blue:wlan5g";
+			gpios = <&gpio0 7 GPIO_ACTIVE_LOW>;
+		};
+		wlan2g {
+			label = "esr600:blue:wlan2g";
+			gpios = <&gpio3 0 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&gpio0 1 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+			debounce-interval = <60>;
+		};
+
+		wps {
+			label = "wps";
+			gpios = <&gpio0 2 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+			debounce-interval = <60>;
+		};
+
+	};
+
+ };
+
+&gpio1 {
+	status = "okay";
+};
+&gpio2 {
+	status = "okay";
+};
+&gpio3 {
+	status = "okay";
+};
+
+&spi0 {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <50000000>;
+		m25p,fast-read;
+	
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+			   
+			partition@0 {
+				label = "u-boot";
+				reg = <0x0 0x30000>;
+				read-only;
+			};
+
+			partition@30000 {
+				label = "u-boot-env";
+				reg = <0x30000 0x10000>;
+				read-only;
+			};
+
+			factory: partition@40000 {
+				label = "factory";
+				reg = <0x40000 0x10000>;
+				read-only;
+			};
+
+			iNIC_rf: partition@50000 {
+				label = "iNIC_rf";
+				reg = <0x50000 0x10000>;
+				read-only;
+			};
+
+			partition@60000 {
+				label = "firmware";
+				reg = <0x60000 0xf40000>;
+				compatible = "denx,uimage";
+			};
+
+			partition@fa0000 {
+				label = "backup";
+				reg = <0xfa0000 0x10000>;
+				read-only;
+			};
+
+			partition@fb0000 {
+				label = "storage";
+				reg = <0xfb0000 0x50000>;
+				read-only;
+			};
+		};
+	};
+};
+
+&ethernet {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&rgmii1_pins &mdio_pins>;
+	mtd-mac-address = <&factory 0x4>;
+	mtd-mac-address-increment = <2>;
+
+        port@5 {
```

Comment:
Some whitespace issue here?

Reply:
Where is the reg property?

Reply:
Doesn't it just merge the new properties into the existing port@5 in the ethernet definition?  Or does it do a complete replacement of the existing port@5 by this (new) port@5 -- in which case I do indeed need everything that's already in the ethernet/port@5 from mt7620a.dtsi, and likewise for the mdio-bus definition, for mdio-bus/#address-cells and #size-cells.

Reply:
I've verified that the dtc compiler does what I expected and merges the nodes under the ethernet node in the mt7620a.dtsi, resulting in subnodes that look as below, do you still require the change?  (I am certainly fixing the whitespace issues though)

		port@4 {
			compatible = "mediatek,mt7620a-gsw-port", "mediatek,eth-port";
			reg = <0x4>;
			status = "disabled";
		};

		port@5 {
			compatible = "mediatek,mt7620a-gsw-port", "mediatek,eth-port";
			reg = <0x5>;
			status = "okay";
			phy-mode = "rgmii";
			mediatek,fixed-link = <0x3e8 0x1 0x1 0x1>;
		};

		mdio-bus {
			#address-cells = <0x1>;
			#size-cells = <0x0>;
			status = "okay";
			mediatek,mdio-mode;

			ethernet-phy@0 {
				reg = <0x0>;
				phy-mode = "rgmii";
				qca,ar8327-initvals = <0x10 0x40000000 0x4 0x7600000 0x7c 0x7e 0xc 0x5600000 0x94 0x7e>;
			};
		};